fendermint/testing/materializer/tests/docker.rs

```diff
@@ -98,14 +94,40 @@ where
         .await
         .context("failed to remove testnet")?;
 
-    let mut testnet = Testnet::setup(&mut materializer, &testnet_name, &manifest)
+    let testnet = Testnet::setup(&mut materializer, &testnet_name, &manifest)
         .await
         .context("failed to set up testnet")?;
 
     let started = wait_for_startup(&testnet).await?;
 
     let res = if started {
-        test(&manifest, &mut materializer, &mut testnet).await
+        match concurrency {
+            None => test(&manifest, &materializer, &testnet, 0).await,
+            Some(cfg) => {
+                let mut futures = Vec::new();
+                let mut test_ids = Vec::new();
+                for i in 0..cfg.parallelism_level {
+                    let test_id = i;
+                    let task = test(&manifest, &materializer, &testnet, test_id);
```

Comment:
just a quick question, the concurrency means running the same test multiple times? Or run multiple different tests at the same time. From the implementation, looks like it's running the same test multiple times?

The `test_id` seems to be account idx?

Reply:
The same test closure is invoked, but each invocation, bound to `test_id`, returns a unique `Future` to be executed concurrently. 

The `test_id` can be used to derive any kind of uniqueness in the scenario, not necessarily with accounts as in this example.

Reply:
by the same test closure, is it running the same test?

Reply:
A test function defines one closure, that can be invoked to define unique tests, which are the same only conceptually. It depends on how `test_id` is used to define it.